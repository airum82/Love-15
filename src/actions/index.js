export const fetchCourts = (nearbyCourts) => ({
  type: 'COURTS',
  nearbyCourts
})

export const createAccount = (accountInfo) => ({
  type: 'CREATE_ACCOUNT',
  accountInfo
})

export const logIn = (accountInfo) => ({
  type: 'LOG_IN',
  accountInfo
})

export const logOut = () => ({
  type: 'LOG_OUT',
})

export const fetchAccount = (accountInfo) => ({
  type: 'FETCH_ACCOUNT',
  email: accountInfo
})