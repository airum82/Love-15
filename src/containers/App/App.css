@import url('https://fonts.googleapis.com/css?family=Merriweather');

body {
  --light-red: rgba(214, 55, 71, .6);
  --court-blue:  rgba(0, 75, 130, 1);
  --harvest:  rgba(229, 149, 0, 1);
  --off-white:  rgba(229, 218, 218, 1);
  --off-black:  rgba(7, 14, 56, 1);
  font-family: 'Merriweather', serif;
}

.App {
  text-align: center;
}

.header-styling {
  margin: 0 auto;
}
