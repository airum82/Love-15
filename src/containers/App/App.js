import React, { Component } from 'react';
import './App.css';
import { Route, NavLink, Link, withRouter } from 'react-router-dom';
import { connect } from 'react-redux';
import MapContainer from '../googleMap/GoogleMap';
import CourtsContainer from '../CourtsContainer/CourtsContainer';
import PropTypes from 'prop-types';
import CreateAccount from '../CreateAccount/CreateAccount';
import LogIn from '../LogIn/LogIn';

export class App extends Component {

  render() {
    return (
      <div className="App">
        <NavLink to='/'>
          <h1>Love-15</h1>
          <h3>It's time to rally!</h3>
        </NavLink>
        <Route exact path = '/' component={LogIn} />
        <Route exact path = '/' component={MapContainer} />
        <Route exact path = '/' component={CourtsContainer} />
        <Route path = '/createAccount' component={CreateAccount} />
      </div>
    );
  }
}

export const mapStateToProps = (state) => ({
  closeCourts: state.closeCourts
})

export default withRouter(connect(mapStateToProps, null)(App));

App.propTypes = {
  closeCourts: PropTypes.array
}