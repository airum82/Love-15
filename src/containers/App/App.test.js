import React from 'react';
import ReactDOM from 'react-dom';
import { App, mapStateToProps, mapDispatchToProps } from './App';
import { shallow } from 'enzyme';
import { fetchAccount } from '../../actions';

describe('App', () => {
  
  it('should match snapshot upon initial render', () => {
    const wrapper = shallow(<App />);
    expect(wrapper).toMatchSnapshot();
  })

  it('mapStateToProps should return an object', () => {
    const mockState = {
      closeCourts: [],
      account: {
        email: 'jim@jim.com'
      }
    }
    const expectedOutput = {
      closeCourts: [],
      account: {
        email: 'jim@jim.com'
      }
    }
    expect(mapStateToProps(mockState)).toEqual(expectedOutput);
  })

  it('fetchUser should call mockDispatch with correct params', () => {
    const mockDispatch = jest.fn();
    const mappedProps = mapDispatchToProps(mockDispatch);
    const mockUser = 'jim@jim.com';
    mappedProps.fetchUser(mockUser);
    expect(mockDispatch).toHaveBeenCalledWith(fetchAccount(mockUser));
  })
})