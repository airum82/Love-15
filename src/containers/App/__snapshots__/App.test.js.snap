// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match snapshot upon initial render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="header-styling"
>
        <Unknown />
        <Route
                component={[Function]}
                exact={true}
                path="/logIn"
        />
        <Route
                exact={true}
                path="/"
                render={[Function]}
        />
        <Route
                component={[Function]}
                path="/createAccount"
        />
        <Connect(CourtMapContainer) />
</div>,
      "className": "App",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Unknown />,
          <Route
            component={[Function]}
            exact={true}
            path="/logIn"
/>,
          <Route
            exact={true}
            path="/"
            render={[Function]}
/>,
          <Route
            component={[Function]}
            path="/createAccount"
/>,
          <Connect(CourtMapContainer) />,
        ],
        "className": "header-styling",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/logIn",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "exact": true,
            "path": "/",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "path": "/createAccount",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="header-styling"
>
          <Unknown />
          <Route
                    component={[Function]}
                    exact={true}
                    path="/logIn"
          />
          <Route
                    exact={true}
                    path="/"
                    render={[Function]}
          />
          <Route
                    component={[Function]}
                    path="/createAccount"
          />
          <Connect(CourtMapContainer) />
</div>,
        "className": "App",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Unknown />,
            <Route
              component={[Function]}
              exact={true}
              path="/logIn"
/>,
            <Route
              exact={true}
              path="/"
              render={[Function]}
/>,
            <Route
              component={[Function]}
              path="/createAccount"
/>,
            <Connect(CourtMapContainer) />,
          ],
          "className": "header-styling",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/logIn",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "exact": true,
              "path": "/",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "/createAccount",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
